structure/set/unionfind.go

```diff
@@ -0,0 +1,63 @@
+package set
+
+// https://en.wikipedia.org/wiki/Disjoint-set_data_structure
+
+type Value interface{}
+
+type node struct {
+	rank  int
+	value Value
+}
+
+type UFS struct {
+	nodes map[Value]*node
+}
+
+func NewUFS() *UFS {
+	return &UFS{
+		nodes: make(map[Value]*node),
+	}
+}
+
+func (u *UFS) Find(v Value) Value {
+	return u.find(v, false).value
+}
+
+func (u *UFS) find(value Value, init bool) *node {
+	p, ok := u.nodes[value]
+	if !ok {
+		p = &node{value: value}
+		if init {
+			u.nodes[value] = p
+		}
+	}
+	if p.value != value {
+		u.nodes[value] = u.find(p.value, init)
+	}
+	return p
+}
+
+func (u *UFS) Connected(x, y Value) bool {
+	return u.Find(x) == u.Find(y)
+}
+
```

Comment:
I'm curious as to what you're doing here? The notion of sets have nothing to do with connectedness as far as my mathematical knowledge goes. Therefore this doesn't belong with this structure.